internal/pkg/webconfig/webconfig.go

```diff
@@ -0,0 +1,296 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package webconfig
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	ttls "github.com/talos-systems/crypto/tls"
+	"github.com/talos-systems/crypto/x509"
+	tnet "github.com/talos-systems/net"
+
+	"github.com/talos-systems/talos/pkg/grpc/gen"
+	"github.com/talos-systems/talos/pkg/machinery/config"
+	"github.com/talos-systems/talos/pkg/machinery/config/configloader"
+)
+
+const indexPage = `
+<html>
+	<head>
+		<title>Talos Interactive Configurator</title>
+	</head>
+	<body>
+	  <h3>Talos Interactive Configuration</h3>
+	  <form action="/" method="POST" enctype="multipart/form-data">
+	  <p>Upload configuration: <input type="file" name="config" /></p>
```

Comment:
Nit: we could call the field config.yaml, so it's easier to upload it from the command line?